models/contacts.js

```diff
@@ -1,19 +1,115 @@
-// const fs = require('fs/promises')
+const { readFile, writeFile } = require("fs").promises;
+const path = require("path");
+const {
+    newContactAuthSchema,
+    editContactAuthSchema,
+} = require("../validation/validation");
 
-const listContacts = async () => {}
+const contactsFile = path.basename("./models/contacts.json");
+const contactsDir = path.dirname("./models/contacts.json");
+const contactsPath = path.join(contactsDir, contactsFile);
 
-const getContactById = async (contactId) => {}
+const listContacts = async () => {
+    return readFile(contactsPath, "utf8")
+        .then((data) => JSON.parse(data))
+        .catch((err) => console.log(err.message));
+};
 
-const removeContact = async (contactId) => {}
+const getContactById = async (contactId) => {
```

Comment:
Warto dodać walidację id przed przeszukaniem listy kontaktów. To może pomóc w obsłudze niepoprawnych lub nieistniejących id

Reply:
jak wyżej. proszę  o wskazówkę.